service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java

```diff
@@ -0,0 +1,213 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static greencity.constant.ErrorMessage.*;
+
+@Service
+@AllArgsConstructor
+public class ViolationServiceImpl implements ViolationService {
+    private ViolationRepository violationRepository;
+    private UserRepository userRepository;
+    private UserViolationsTableRepo userViolationsTableRepo;
+    private OrderRepository orderRepository;
+
+    private final EventService eventService;
+    private final NotificationServiceImpl notificationService;
+    private final FileService fileService;
+
+    /**
+     * Constructor with parameters.
+     */
+    @Autowired
+    public ViolationServiceImpl(EventService eventService, NotificationServiceImpl notificationService,
+        FileService fileService) {
+        this.eventService = eventService;
+        this.notificationService = notificationService;
+        this.fileService = fileService;
+    }
+
+    @Override
+    public UserViolationsWithUserName getAllViolations(Pageable page, Long userId, String columnName,
+        SortingOrder sortingOrder) {
+        String username = getUsername(userId);
+        Long numberOfViolations = violationRepository.getNumberOfViolationsByUser(userId);
+
+        Page<Violation> violationPage = userViolationsTableRepo.findAll(userId, columnName, sortingOrder, page);
+
+        List<UserViolationsDto> userViolationsList =
+            violationPage.getContent()
+                .stream()
+                .map(this::getAllViolations)
+                .collect(Collectors.toList());
+
+        return UserViolationsWithUserName.builder()
+            .userViolationsDto(new PageableDto<>(userViolationsList, numberOfViolations,
+                violationPage.getPageable().getPageNumber(), violationPage.getTotalPages()))
+            .fullName(username)
+            .build();
+    }
+
+    private UserViolationsDto getAllViolations(Violation violation) {
+        return UserViolationsDto.builder()
+            .violationDate(violation.getViolationDate())
+            .orderId(violationRepository.getOrderIdByViolationId(violation.getId()))
+            .violationLevel(violation.getViolationLevel())
+            .build();
+    }
+
+    private String getUsername(Long userID) {
+        User currentUser = userRepository.getOne(userID);
+        return currentUser.getRecipientName() + " " + currentUser.getRecipientSurname();
+    }
+
+    @Override
+    public void addUserViolation(AddingViolationsToUserDto add, MultipartFile[] multipartFiles, String uuid) {
```

Comment:
these parameters should be refactored. uuid - not clear what kind of UUId it is. Also, all of these parameters can be a part of dto, instead of separate fields. you can set multipartFiles as a field of dto in upper class level.